mlir/lib/Dialect/Rock/Transforms/GridwiseGemmToBlockwise.cpp

```diff
@@ -138,6 +138,24 @@ computeCopyPerThread(GemmDimension dim, int64_t vectorLen,
   return std::make_pair(copyKPerThread, copyDPerThread);
 }
 
+static FailureOr<std::pair<int64_t, int64_t>>
+computeCopyPerThreadV2(int64_t vectorLen, int64_t copyPerThread,
+                       int64_t kPerBlock, int64_t dPerBlock, Location loc) {
+  int64_t copyKPerThread = std::min(vectorLen, copyPerThread);
```

Comment:
Ok, to rephrase my comment, I don't think it's good to always have the k dimension be the vectorization dimension. The optimal vectorization for LDS is either K or M, depending on whethere kPack == 1 or not

Unless I've really misunderstood your changes

Reply:
> The optimal vectorization for LDS is either K or M, depending on whethere kPack == 1 or not

Agreed, so we should statically pick the optimal dimension (based on kPack) rather than make it based on potential vectorization length.